client/tinychain/ml/nn.py

```diff
@@ -0,0 +1,255 @@
+import functools
+import operator
+from typing import List
+
+from tinychain.collection.tensor import einsum, Dense, Tensor
+from tinychain.ml import Layer, Identity, Sigmoid, Tanh, ReLU, DiffedParameter, Parameter, NeuralNet
+from tinychain.ref import After
+
+
+class DNNLayer(Layer):
+    @classmethod
+    def create(cls, name: str, input_size, output_size, activation=Identity()):
+        """Create a new, empty `DNNLayer` with the given shape and activation function.
+        Args:
+            `name`: name and number of layer;
+            `input_size`: size of inputs;
+            `output_size`: size of outputs;
+            `activation`: activation function.
+        """
+
+        return {'name': name, 'input_size': input_size, 'output_size': output_size, 'activation': activation}
+
+    @classmethod
+    def initialize_layer(cls, name: str, input_size, output_size, activation=Identity()):
+        """Initializing `ConvLayer` parameters by Xavier initialization for Sigmoid, Tanh activations,
+        and Kaiming initialization for ReLU
+        """
+
+        if activation == Identity():
```

Comment:
This logic should go in a method in `Activation` itself so that it will work for user-defined activation functions

Reply:
Like maybe `Activation.initializer`